pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -1035,12 +1037,25 @@ func encodeNetworkPortStatus(ctx *zedagentContext,
 	devicePort.Up = port.Up
 	devicePort.Mtu = uint32(port.MTU)
 	devicePort.Domainname = port.DomainName
-	// TODO: modify EVE APIs and allow to publish full list of NTP servers
-	if port.NtpServer != nil {
-		devicePort.NtpServer = port.NtpServer.String()
-	} else if len(port.NtpServers) > 0 {
-		devicePort.NtpServer = port.NtpServers[0].String()
+	ntpServers := make([]string, 0)
+	if port.ConfiguredNtpServers != nil {
+		ntpServers = append(ntpServers, port.ConfiguredNtpServers...)
+	}
+	if len(port.DhcpNtpServers) > 0 && !port.IgnoreDhcpNtpServers {
+		devicePort.NtpServer = port.DhcpNtpServers[0].String()
```

Comment:
You seem to be overwriting this devicePort.NtpServer on line 1053.  Isn't the intent to collect all the ntp servers in the ntpServers variable and then update the devicePort below?

Reply:
Yes, this code is wrong. It should be fixed now.